### ISSUE TITLE
Assignment methods don't work if there is no underlying variable

### ISSUE THREAD

**Issue by KimBruce** [gracelang/minigrace#110]
Consider the following code:

```
def o = object {
    var hx: Number := 0
    
    method x:= (n: Number) -> Done {
        hx := n
    }
    
    self.x := 6
    
    method report {
        print(hx)
    }
}
o.report
```
-----------
This works fine, but if you replace `self.x := 6` by `x := 6` (i.e., just erase "self.")  you get the following error message:
Syntax error: Unknown variable or method 'x'. This may be a spelling mistake or an attempt to access a variable in another scope.
I can see this being tricky for the parser to get right, but ...

**Comment by kjx** [comment]
> On 17/06/2015, at 9:04am, Kim Bruce <notifications@github.com> wrote:
> 
> I can see this being tricky for the parser to get right, but …

sounds like an amg bug.  Runs fine on kernan and my old mg build

James

**Comment by apblack** [comment]
This is fixed in commit 56971180cd623a67c1f83f7b6ae7e997415435a0.   I was checking for variables and methods in the wrong order.
